src/AttestationData.ts

```diff
@@ -0,0 +1,408 @@
+    generality?: number
+
+    // ...extensible with other fields that summarize the content of the attestation - e.g., a list of addresses, accounts, totals of statistics, etc.
+  }
+
+  // Core attestation data, dependent on attestation type
+  data?: Array<Object | string | number> | Object | string | number
+  // ...extensible with other fields.  Other fields explicating general data about the attestation, such as location, shelf life, common units, etc., should be placed here.
+}
+
+///////////////////////////////////////////////////
+// Helper types
+///////////////////////////////////////////////////
+export type TPersonalName =  // Designed to be flexible - as a rule, a basic {given: "x", middle: "x", family: "x"} is probably the easiest for most Western use cases
+  | string
+  | {
+      full?: string
+      given?: string | Array<string>
+      middle?: string | Array<string>
+      family?: string | Array<string>
+      title?: string | Array<string>
+      prefix?: string | Array<string>
+      suffix?: string | Array<string>
+      nickname?: string | Array<string>
+      generational?: string | Array<string>
+
+      // For name changes
+      start_date?: TDate
+      end_date?: TDate
+    }
+
+export type TDate =
+  | string // ISO-8601 date in YYYY-MM-DD format
+  | {
+      year: string // YYYY
+      month: string // MM
+      day: string // DD
+    }
+
+export type TDatetime =
+  | string // ISO-8601 datetime in YYYY-MM-DDTHH:MM:SSZ format
+  | {
+      year: string // YYYY
+      month: string // MM
+      day: string // DD
+      hour: string // HH
+      minute: string // MM
+      second: string // SS
+    }
+
+export type TPhoneNumber =
+  | string // Valid internationally-formatted phone number
+  | {
+      full?: string
+      country?: string
+      subscriber?: string
+      area?: string
+      prefix?: string
+      line?: string
+      ext?: string
+    }
+
+export type TGender = string // 'male', 'female', ...
+
+export type TAddress = {
+  full: string
+  name: string
+  street_1: string
+  street_2?: string
+  street_3?: string
+  city: string
+  postal_code: string | number
+  region_1: string
+  region_2?: string
+  country?: string
+}
+
+///////////////////////////////////////////////////
+// Phone attestation dataStr type
+///////////////////////////////////////////////////
+export type TBaseAttPhoneData = string | TPhoneNumber
+export interface IBaseAttPhone extends IBaseAtt {
+  data: TBaseAttPhoneData | Array<TBaseAttPhoneData>
+}
+
+///////////////////////////////////////////////////
+// Email attestation dataStr type
+///////////////////////////////////////////////////
+export interface IBaseAttEmailData {
+  email?: string
+  start_date?: string
```

Comment:
Why does this one have `start/end_date`

Reply:
Sometimes emails associated with an individual or organization have a beginning or expiry date.  Just so happens I had that exact problem this morning.  E.g., a company issuing an attestation for when an employee has access to an email.